File: Codeforces/230B - T-primes.cpp
#include <iostream>
#include <cmath>

using namespace std;

#define endl '\n'
#define fast                      \
    ios_base::sync_with_stdio(0); \
    cin.tie(0);                   \
    cout.tie(0);

typedef unsigned long long ull;

bool is_prime(ull n)
{
    if (n == 1 or n % 2 == 0)
        return false;
    else if (n == 2)
        return true;
    else
        for (ull i = 3; i * i <= n; i += 2)
            if (n % i == 0)
                return false;
    return true;
}

int main()
{
    fast;
    int n;
    cin >> n;
    while (n--)
    {
        ull a;
        cin >> a;
        ull s = sqrt(a);
        if (s * s == a and is_prime(s))
            cout << "YES" << endl;
        else
            cout << "NO" << endl;
    }

    return 0;
}

File: Codeforces/230B - T-primes.py
n = int(input())
li = list(map(int, input().split()))


def is_prime(a):
    if a == 2:
        return True
    elif a == 1 or a % 2 == 0:
        return False
    else:
        for i in range(3, int(a ** 0.5), 2):
            if a % i == 0:
                return False
    return True


def chck(a):
    s = int(a ** 0.5)
    if s * s == a and is_prime(s):
        return "YES"
    else:
        return "NO"


for l in li:
    print(chck(l))
